src/algebra/category/Module/subobject.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2021 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import algebra.category.Module.basic
+import category_theory.subobject.basic
+
+/-!
+# Subobjects in the category of `R`-modules
+
+We construct an explicit order isomorphism between the categorical subobjects of an `R`-module `M`
+and its submodules.
+
+## Future work
+
+Once we have well-powered categories, our calculation immediately implies that `Module.{v} R` is
+well-powered.
+-/
+
+open category_theory
+open category_theory.subobject
+open_locale Module
+
+universes v u
+
+namespace Module
+variables {R : Type u} [ring R] (M : Module.{v} R)
+
+/-- The categorical subobjects of a module `M` are in one-to-one correspondence with its
+    submodules.-/
+noncomputable def subobject_Module : subobject M ≃o submodule R M := order_iso.symm
```

Comment:
Nice! I struggled to prove the order_iso directly in #6812, and resorted to building the equivalence from `mono_over M`.